lib/pact/consumer_contract/request_decorator.rb

```diff
@@ -20,25 +20,35 @@ def as_json options = {}
       hash[:query]   = query   if request.specified?(:query)
       hash[:headers] = headers if request.specified?(:headers)
       hash[:body]    = body    if request.specified?(:body)
-      hash
+      include_matching_rules? ? with_matching_rules(hash) : hash
     end
 
     private
 
     attr_reader :request
 
     def path
```

Comment:
Can you include a test please? I understand that there may not be an existing test, however, a good PR should always leave code in a better state than it started ;)